src/aeso_ast_infer_types.erl

```diff
@@ -2329,6 +2368,33 @@ check_bytes_constraint(Env, {add_bytes, Ann, Fun, A0, B0, C0}) ->
         _ -> type_error({unsolved_bytes_constraint, Ann, Fun, A, B, C})
     end.
 
+%% -- Typevars constraints --
+
+check_tvars_constraints(Env, Constraints) ->
+    [ check_tvars_constraint(Env, C) || C <- Constraints ].
+
+check_tvars_constraint(Env, {is_eq, Type = {uvar, Ann, _}}) ->
+    Type1 = unfold_types_in_type(Env, instantiate(Type)),
+    type_is_eq(Type1) orelse type_error({type_not_eq, Ann, Type1});
+check_tvars_constraint(Env, {is_ord, Type = {uvar, Ann, _}}) ->
+    Type1 = unfold_types_in_type(Env, instantiate(Type)),
+    type_is_ord(Type1) orelse type_error({type_not_ord, Ann, Type1}).
+
+type_is_ord({app_t, _, Id, Ts}) -> type_is_ord(Id) andalso lists:all(fun type_is_ord/1, Ts);
+type_is_ord({tuple_t, _, Ts}) -> lists:all(fun type_is_ord/1, Ts);
+type_is_ord({bytes_t, _, _}) -> true;
+type_is_ord({constrained_t, _, Constraints, {tvar, _, _}}) -> lists:keyfind("ord", 3, Constraints) =/= false;
+type_is_ord({id, _, Id}) -> ets_lookup(ord_constraint_types, Id) =/= [];
+type_is_ord(_) -> false.
```

Comment:
After some thinking I think comparing addresses makes sense. It can be useful if you keep a list of addresses and want to ensure the order by sorting it

Reply:
Done here fc2875965e302c75a1c02e9655beeb0c524aecee

Reply:
Will have to modify the tests for that.

Reply:
Fixed the tests here 4562a7166cfe7ddac65a940508a8266a0e7034f8